spring-core/src/main/java/org/springframework/core/CoroutinesUtils.java

```diff
@@ -71,7 +75,9 @@ public static <T> Deferred<T> monoToDeferred(Mono<T> source) {
 	public static Publisher<?> invokeSuspendingFunction(Method method, Object target, Object... args) {
 		KFunction<?> function = Objects.requireNonNull(ReflectJvmMapping.getKotlinFunction(method));
 		KClassifier classifier = function.getReturnType().getClassifier();
-		Mono<Object> mono = MonoKt.mono(Dispatchers.getUnconfined(), (scope, continuation) ->
+		Continuation<?> cont = (Continuation<?>) args[args.length - 1];
```

Comment:
This might casuse an NPE, and this code is validated
```java
public static Publisher<?> invokeSuspendingFunction(Method method, Object target, Object... args) {
		KFunction<?> function = Objects.requireNonNull(ReflectJvmMapping.getKotlinFunction(method));
		if (method.isAccessible() && !KCallablesJvm.isAccessible(function)) {
			KCallablesJvm.setAccessible(function, true);
		}
		KClassifier classifier = function.getReturnType().getClassifier();
		Continuation<?> originContinuation = (Continuation<?>) args[args.length - 1];
		CoroutineContext context;
		if (originContinuation == null) {
			context = (CoroutineContext) Dispatchers.getUnconfined();
		} else {
			context=originContinuation.getContext().minusKey(Job.Key);
		}
		Mono<Object> mono = MonoKt.mono(context, (scope, continuation) ->
					KCallables.callSuspend(function, getSuspendedFunctionArgs(target, args), continuation))
				.filter(result -> !Objects.equals(result, Unit.INSTANCE))
				.onErrorMap(InvocationTargetException.class, InvocationTargetException::getTargetException);
		if (classifier != null && classifier.equals(JvmClassMappingKt.getKotlinClass(Flow.class))) {
			return mono.flatMapMany(CoroutinesUtils::asFlux);
		}
		return mono;
	}
```

Reply:
Thanks for pointing this out 👍.

Reply:
I don't think the refined versioned shared below is impacted since we are always in a case where there is a `Continuation` parameter, but feel free to double check.